coordinator/src/routes/dispense/verify-signature.ts

```diff
@@ -5,84 +5,125 @@ import {
+//  1b. FHIR Acute/Repeat Prescribing/Repeat Dispensing prescriptions
+//  1c. No prescription in tracker for prescription id(s) in request
+// 2. Error handling for no prescription found in tracker
 
 export default [
-  /*
-      Verify prescription bundle signatures.
-    */
   {
     method: "POST",
     path: `${BASE_PATH}/$verify-signature`,
     handler: externalValidator(
-      async (request: Hapi.Request, responseToolkit: Hapi.ResponseToolkit) => {
-        const outerBundle = getPayload(request) as fhir.Resource
-        if (!isBundle(outerBundle)) {
+      async (request: Hapi.Request, responseToolkit: Hapi.ResponseToolkit): Promise<Hapi.ResponseObject> => {
+        const bundle = getPayload(request) as fhir.Resource
+        if (!isBundle(bundle)) {
           const operationOutcome = fhir.createOperationOutcome([
             errors.createResourceTypeIssue("Bundle")
           ])
           return responseToolkit.response(operationOutcome).code(400).type(ContentTypes.FHIR)
         }
-        request.logger.info("Verifying prescription signatures from Bundle")
-        const verificationResponses = outerBundle.entry
-          .map(entry => entry.resource)
-          .filter(isBundle)
-          .map((innerBundle, index) => verifyPrescriptionSignature(innerBundle, index, request.logger))
 
-        const parameters: fhir.Parameters = {
-          resourceType: "Parameters",
-          parameter: verificationResponses
+        const prescriptions = isReleaseResponse(bundle)
+          ? getBundlesFromReleaseResponse(bundle)
+          : isBundle(bundle)
+            ? toArray(bundle)
+            : null
+
+        if (prescriptions === null) {
+          request.logger.error("Did not receive a release response or FHIR bundle prescription")
+          // todo: return error response
         }
 
-        return responseToolkit.response(parameters).code(200).type(ContentTypes.FHIR)
+        request.logger.info("Verifying prescription(s)")
+
+        const parameters = await Promise.all(
+          prescriptions.map(async(fhirPrescriptionFromRequest: fhir.Bundle, index: number) => {
+            const fhirPathBuilder = new common.FhirPathBuilder()
+            const fhirPathReader = new common.FhirPathReader(fhirPrescriptionFromRequest)
+            const medicationRequest = fhirPathBuilder.bundle().medicationRequest()
+            const prescriptionId = fhirPathReader.read(medicationRequest.prescriptionShortFormId())
+            const repeatNumber = getRepeatNumber(fhirPathReader.read(medicationRequest.repeatsIssued()))
+            const trackerResponse = await trackerClient.track(
+              getRequestId(request.headers),
+              prescriptionId,
+              repeatNumber,
+              request.logger
+            )
+            // todo: handle errors inc. no prescription returned
+            const hl7v3PrescriptionFromTracker = trackerResponse.prescription
```

Comment:
suggest writing a v3Builder/v3Reader to compare values in the two prescriptions. think it will make the code more maintainable for people who don't know what is going on at this point. not one for this pr